aeron-driver/src/main/java/io/aeron/driver/DriverConductor.java

```diff
@@ -1598,7 +1598,8 @@ private NetworkPublication newNetworkPublication(
                 ctx.senderCachedNanoClock(),
                 ctx.systemCounters().get(INVALID_PACKETS),
                 ctx.retransmitUnicastDelayGenerator(),
-                ctx.retransmitUnicastLingerGenerator());
+                ctx.retransmitUnicastLingerGenerator(),
+                udpChannel.isMulticast());
```

Comment:
We use to have a variable or function called `hasGroupSemantics` which I think got dropped during the previous iteration through the RetransmitHandler.  I think we need to reintroduce that as we should only be using the unicast semantics if it actually unicast.  Specifically we need to treat MDC like it is multicast for retransmits and we should also take into account the `group=true` URI parameter.